Add TavilyExtractorTool and TavilySearchTool with documentation

Programmer-RD-AI opened the pull request:
Introduce two new tools, `TavilyExtractorTool` and `TavilySearchTool`, enabling CrewAI agents to extract content and perform searches using the Tavily API. Include installation instructions and usage examples in the documentation. Update dependencies to include `tavily-python`.

Fixes #278

lorenzejay commented:
results are blowing up context window.

<img width="667" alt="Screenshot 2025-05-13 at 12 41 49 PM" src="https://github.com/user-attachments/assets/f4d3bec3-5c3d-438d-9ded-b79c0397e5ab" />

lorenzejay commented on crewai_tools/tools/tavily_search_tool/tavily_search_tool.py:
```diff
@@ -0,0 +1,213 @@
+    def _run(
+        self,
+        query: str,
+        search_depth: Literal["basic", "advanced"] = "basic",
+        topic: Literal["general", "news", "finance"] = "general",
+        time_range: Optional[Literal["day", "week", "month", "year"]] = None,
+        days: int = 7,
+        max_results: int = 5,
+        include_domains: Optional[Sequence[str]] = None,
+        exclude_domains: Optional[Sequence[str]] = None,
+        include_answer: Union[bool, Literal["basic", "advanced"]] = False,
+        include_raw_content: bool = False,
+        include_images: bool = False,
+        timeout: int = 60,
```
i'd recommend moving all but query to init vs on run. wdyt

lorenzejay commented on crewai_tools/tools/tavily_extractor_tool/tavily_extractor_tool.py:
```diff
@@ -0,0 +1,156 @@
+        else:
+            import click
+
+            if click.confirm(
+                "The 'tavily-python' package is required to use the TavilyExtractorTool. "
+                "Would you like to install it?"
```
"You are missing the 'tavily-python' package. Would you like to install it? "

lorenzejay commented on crewai_tools/tools/tavily_extractor_tool/tavily_extractor_tool.py:
```diff
@@ -0,0 +1,156 @@
+    async def _arun(
+        self,
+        urls: Union[List[str], str],
+        include_images: bool = False,
+        extract_depth: Literal["basic", "advanced"] = "basic",
```
laso move to init

lorenzejay commented:
this is how i ran it and it blew up:

```python
if __name__ == "__main__":
    tool = TavilySearchTool()

    agent = Agent(
        role="Tavily Search Agent",
        goal="Search the web for information about the capital of France",
        backstory="You are a helpful assistant that can search the web for information.",
        tools=[tool],
        verbose=True,
    )
    task = Task(
        description="Search the web for information about the capital of France",
        expected_output="A JSON object containing the search results.",
        agent=agent,
    )
    crew = Crew(
        agents=[agent],
        tasks=[task],
    )
    crew.kickoff()
```

Programmer-RD-AI commented:
Hey @lorenzejay,

Thanks for your thorough review and feedback on PR #279. I've addressed your suggestions as follows:

1. **Parameter Initialization**: Moved all parameters except `query` to the `__init__` method for both `TavilySearchTool` and `TavilyExtractorTool`, ensuring cleaner and more efficient instantiation.

2. **Prompt Message Update**: Updated the prompt message for missing dependencies to:

   > "You are missing the 'tavily-python' package. Would you like to install it?"

3. **Context Window Management**: Adjusted the tools to manage the context window more effectively, preventing it from being overwhelmed during execution.

These changes are reflected in the latest commit [dedbd8c](https://github.com/crewAIInc/crewAI-tools/commit/dedbd8c).

Could you please review the updates and let me know if they resolve the issues you encountered? I'm open to further suggestions to ensure seamless integration.

Best regards,
Programmer-RD-AI

lorenzejay reviewed (approved):
a lot better! great work @Programmer-RD-AI

lucasgomide reviewed (approved):
great job @Programmer-RD-AI

Programmer-RD-AI commented:
Note: I’ve opened a companion PR in the main crewAI repo to add full documentation for these tools, see https://github.com/crewAIInc/crewAI/pull/3146. Let me know if you’d like any tweaks on the docs side.

Programmer-RD-AI commented:
Opened a companion PR #378 to export `TavilySearchTool` and `TavilyExtractorTool` in `crewai_tools/__init__.py`. This lets you do:

```py
from crewai_tools import TavilySearchTool, TavilyExtractorTool
```

No breaking changes.
